### PR TITLE
CLIENTS-1597: Add relationship reassignment and refactor reassignment resource

### PR DESCRIPTION
**Pull request by agg111**
### What
- Added relationship for reassignment to cluster, topic and partition.
- Refactor `reassignments` to `reassignment` since it is indeed a resource not a collection.

### Testing
Modified unit and integration tests.
Tested locally with `mvn test`

### CODE REVIEW

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java [hunk 1]
```diff
@@ -54,6 +54,9 @@
   @JsonProperty("topic_configs")
   public abstract Relationship getTopicConfigs();
 
+  @JsonProperty("reassignment")
```
Comment (rigelbm) [inline kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java:57]: partition_reassignments

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/TopicData.java [hunk 2]
```diff
@@ -44,6 +44,9 @@
   @JsonProperty("configs")
   public abstract Relationship getConfigs();
 
+  @JsonProperty("reassignment")
```
Comment (rigelbm) [inline kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/TopicData.java:47]: partition_reassignments

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java [hunk 3]
```diff
@@ -71,6 +74,7 @@ static ClusterData fromJson(
       @JsonProperty("acls") Relationship acls,
       @JsonProperty("brokers") Relationship brokers,
       @JsonProperty("broker_configs") Relationship brokerConfigs,
+      @JsonProperty("reassignment") Relationship reassignment,
```
Comment (rigelbm) [inline kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java]: partition_reassignments

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java [hunk 4]
```diff
@@ -54,6 +54,9 @@
   @JsonProperty("topic_configs")
   public abstract Relationship getTopicConfigs();
 
+  @JsonProperty("partition_reassignments")
+  public abstract Relationship getReassignment();
```
Comment (rigelbm) [inline kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java:58]: partitionReassignments()

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java [hunk 5]
```diff
@@ -111,6 +116,8 @@ static ClusterData fromJson(
 
     public abstract Builder setTopicConfigs(Relationship topicConfigs);
 
+    public abstract Builder setReassignment(Relationship reassignment);
```
Comment (rigelbm) [inline kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java:119]: setPartitionReassignments(Relationship partitionReassignments);

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/TopicData.java [hunk 6]
```diff
@@ -65,7 +68,8 @@ static TopicData fromJson(
       @JsonProperty("is_internal") boolean isInternal,
       @JsonProperty("replication_factor") int replicationFactor,
       @JsonProperty("partitions") Relationship partitions,
-      @JsonProperty("configs") Relationship configs
+      @JsonProperty("configs") Relationship configs,
+      @JsonProperty("reassignment") Relationship reassignment
```
Comment (rigelbm) [inline kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/TopicData.java]: partition_reassignments

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/TopicData.java [hunk 7]
```diff
@@ -97,6 +102,8 @@ static TopicData fromJson(
 
     public abstract Builder setConfigs(Relationship configs);
 
+    public abstract Builder setReassignment(Relationship reassignment);
```
Comment (rigelbm) [inline kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/TopicData.java:105]: setPartitionReassignments

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java (hunk 1 above)
Comment (agg111) [reply kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java]: Modified to `partition_reassignments`

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/TopicData.java (hunk 2 above)
Comment (agg111) [reply kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/TopicData.java]: Updated to `partition_reassignments`

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java (hunk 3 above)
Comment (agg111) [reply kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java]: Changed to `partition_reassignments`

File: kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java (hunk 5 above)
Comment (agg111) [reply kafka-rest/src/main/java/io/confluent/kafkarest/entities/v3/ClusterData.java]: Modified all getters and setters for ClusterData and TopicData classes and tests.

**Pull request merged**